src/formatDate.js

```diff
@@ -50,7 +50,36 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  const dateParts = date.split(fromSeparator);
+
+  const mapping = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    mapping[fromFormat[i]] = dateParts[i];
+  }
+
+  if (mapping['YY'] && !mapping['YYYY']) {
+    if (parseInt(mapping['YY']) < 30) {
+      mapping['YYYY'] = '20' + mapping['YY'];
```

Comment:
suggest to make constants outside of function instead hardcoded strings for `YY` and `YYYY` and use in all places